src/js/ui/simply-pebble.js

```diff
@@ -1113,6 +1132,35 @@ SimplyPebble.accelConfig = function(def) {
   SimplyPebble.sendPacket(AccelConfigPacket.prop(def));
 };
 
+SimplyPebble.voiceDictationSession = function(callback) {
+  // If there's a transcription in progress
+  if (SimplyPebble.dictationCallback) {
+    callback( { 'status': -1, 'transcription': null } );
+    return;
+  }
+
+  // Grab the current window to re-show once we're done
+  SimplyPebble.window = WindowStack.top();
+
+  // Set the callback and send the packet
+  SimplyPebble.dictationCallback = callback;
+  SimplyPebble.sendPacket(VoiceDictationStartPacket);
+}
+
+SimplyPebble.onVoiceData = function(packet) {
+  if (!SimplyPebble.dictationCallback) {
+    // Something bad happened
+    console.log("No callback specified for dictation session");
+  } else {
+    // invoke and clear the callback
+    SimplyPebble.dictationCallback( { 'status': packet.status(), 'transcription': packet.transcription() } );
+    SimplyPebble.dictationCallback = null;
+  }
```

Comment:
accelPeek is a little weird in how it handles listeners since it tries to do a best effort of throwing the most recent accel data it was given to all listeners.

For here, you would just get the callback info with `var callbackInfo = dictionationListenerInfos.shift()` and then resume your normal handling as you have here.